attribution-data-file-share/src/main/java/gov/cms/ab2d/attributionDataShare/AttributionDataShareHandler.java

```diff
@@ -0,0 +1,87 @@
+package gov.cms.ab2d.attributionDataShare;
+
+import com.amazonaws.services.lambda.runtime.Context;
+import com.amazonaws.services.lambda.runtime.LambdaLogger;
+import com.amazonaws.services.lambda.runtime.RequestStreamHandler;
+import gov.cms.ab2d.lambdalibs.lib.PropertiesUtil;
+
+import java.io.*;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Properties;
+
+public class AttributionDataShareHandler implements RequestStreamHandler {
+
+    private static final String FILE_PATH = "/opt/";
+    private static final String FILE_PARTIAL_NAME = "ab2d-beneids_";
+    private static final String FILE_FORMAT = ".txt";
+
+    private static final String SELECT_ALL_FROM_COVERAGE = "SELECT DISTINCT current_mbi FROM public.coverage";
+
+    // Writes out a file to the FILE_PATH.
+    // I.E: "ab2d-beneids_2023-08-16T12:08:56.235-0700.txt"
+    public void handleRequest(InputStream inputStream, OutputStream outputStream, Context context) throws IOException {
+        LambdaLogger logger = context.getLogger();
+        logger.log("AttributionDataShare Lambda is started");
+
+        String currentDate = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSZ").format(new Date());
+        String fileFullPath = FILE_PATH + FILE_PARTIAL_NAME + currentDate + FILE_FORMAT;
+
+        try (FileWriter fileWriter = new FileWriter(fileFullPath)) {
+            Connection dbConnection = getConnection();
+
+            Statement statement = dbConnection.createStatement();
+            ResultSet resultSet = statement.executeQuery(SELECT_ALL_FROM_COVERAGE);
```

Comment:
resultSet will be huge if you run the query on prod. This will throw Out of memory exception

Reply:
Is there a way to introduce a buffer? Or maybe query in batches? How can we avoid this out of memory exception?

Reply:
https://stackoverflow.com/questions/1107732/how-to-avoid-oom-out-of-memory-error-when-retrieving-all-records-from-huge-tab

or use cursor like in ab2d for coverage table